src/utils/raw.rs

```diff
@@ -206,7 +206,7 @@ fn symlink_junction_inner(target: &Path, junction: &Path) -> io::Result<()> {
         );
 
         let mut data = [0u8; MAXIMUM_REPARSE_DATA_BUFFER_SIZE];
-        let db = data.as_mut_ptr() as *mut REPARSE_MOUNTPOINT_DATA_BUFFER;
+        let db = data.as_mut_ptr().cast::<REPARSE_MOUNTPOINT_DATA_BUFFER>();
         let buf = &mut (*db).ReparseTarget as *mut WCHAR;
```

Comment:
Why not also do this one?

Reply:
`pointer::cast` can only be applied to pointer type. In this case `&mut (*db).ReparseTarget` is a reference.

Reply:
Aha, I understand.